pkg/ccl/changefeedccl/changefeed_stmt.go

```diff
@@ -763,13 +767,43 @@ func validateDetails(details jobspb.ChangefeedDetails) (jobspb.ChangefeedDetails
 		}
 	}
 	{
-		_, withInitialScan := details.Opts[changefeedbase.OptInitialScan]
+		initialScan, initialScanSet := details.Opts[changefeedbase.OptInitialScan]
 		_, noInitialScan := details.Opts[changefeedbase.OptNoInitialScan]
-		if withInitialScan && noInitialScan {
+		_, onlyInitialScan := details.Opts[changefeedbase.OptInitialScanOnly]
+
+		if initialScanSet && noInitialScan {
+			return jobspb.ChangefeedDetails{}, errors.Errorf(
+				`cannot specify both %s and %s`, changefeedbase.OptInitialScan,
+				changefeedbase.OptNoInitialScan)
+		}
+
+		if initialScanSet && onlyInitialScan {
 			return jobspb.ChangefeedDetails{}, errors.Errorf(
 				`cannot specify both %s and %s`, changefeedbase.OptInitialScan,
+				changefeedbase.OptInitialScanOnly)
+		}
+
+		if noInitialScan && onlyInitialScan {
+			return jobspb.ChangefeedDetails{}, errors.Errorf(
+				`cannot specify both %s and %s`, changefeedbase.OptInitialScanOnly,
 				changefeedbase.OptNoInitialScan)
 		}
+
+		// if the user specified a type for the initial scan option
+		if initialScanSet && initialScan != `` {
+			const opt = changefeedbase.OptInitialScan
+			switch v := changefeedbase.InitialScanOptType(details.Opts[opt]); v {
+			case changefeedbase.OptInitialScanType:
+				details.Opts[opt] = string(changefeedbase.OptInitialScanType)
+			case changefeedbase.OptNoInitialScanType:
+				details.Opts[opt] = string(changefeedbase.OptNoInitialScanType)
+			case changefeedbase.OptInitialScanOnlyType:
+				details.Opts[opt] = string(changefeedbase.OptInitialScanOnlyType)
+			default:
+				return jobspb.ChangefeedDetails{}, errors.Errorf(
+					`unknown %s: %s`, opt, v)
+			}
```

Comment:
I wonder if we should just change `initialScanFromOptions` methods to return InitialScan and an error.  Do some of the checking there... Maybe all of it, actually.

Reply:
yeah I think that might be a better option. Originally I didn't give it much thought because we always perform `validateDetails` before looking at this value, but these validations best belong in the function call.